python/llm/src/bigdl/llm/transformers/speculative.py

```diff
@@ -144,8 +144,10 @@ def _prepare_past_key_values_storage_cpu(self, past_key_values,
         if not _enable_ipex:
             len0 = past_key_values[i][0].size(0)
             len1 = past_key_values[i][0].size(1)
-            len2 = past_key_values[i][0].size(2)
-            len3 = past_key_values[i][0].size(3)
+            # gpt_bigcode has only 2-dimension kv
+            if self.config.model_type != "gpt_bigcode":
+                len2 = past_key_values[i][0].size(2)
+                len3 = past_key_values[i][0].size(3)
```

Comment:
Maybe check the shape of `past_key_values` instead of the `model_type` here?

Reply:
tested and updated